docs/lessons/dance-party.md

```diff
@@ -0,0 +1,511 @@
+
+Now go into ``||sprites:Sprite||``, select ``||sprites:destroy sprite with effect||``, and drag it into on ``||scene:sprite of kind hits wall||``. Then select the effect to be fire and click the plus arrow next to it. Set the time to 100 ms. Drag ``||info:change life by||`` and put it under destroy sprite with effect. Set the number to -1. Now whenever a projectile collides with the bottom it will be destroyed and the player's remaining lives will go down.
+
+![Player moving on the screen](/static/lessons/dance-party/missedcollision.gif)
+
+```blocks
+scene.onHitTile(SpriteKind.Projectile, 1, function (sprite) {
+    sprite.destroy(effects.fire, 100)
+    info.changeLifeBy(-1)
+})
+```
+
+When the player collides with a projectile, their score should increase and the projectile should be destroyed. Go to ``||sprites:Sprites||`` and grab ``||sprites:on sprite of kind player overlaps otherSprite of kind||`` and drag it into your workspace. Change the field after otherSprite from player to projectile. Now go back into sprites and drag ``||sprites:destroy sprite with effect||`` into it. Set the effect to be disintegrate and then select the plus arrow and enter 100 ms. Now go to info and grab ``||info:change score by||`` and drag it under destroy sprite with effect. 
+
+![Player moving on the screen](/static/lessons/dance-party/collisionprojectiles.gif)
+
+```blocks
+sprites.onOverlap(SpriteKind.Player, SpriteKind.Projectile, function (sprite, otherSprite) {
+    otherSprite.destroy(effects.disintegrate, 100)
+    info.changeScoreBy(1)
+})
+```
+
+
+And now we have a fully functioning game!
+
+## Final Game
+
+```blocks
+sprites.onOverlap(SpriteKind.Player, SpriteKind.Projectile, function (sprite, otherSprite) {
+    otherSprite.destroy(effects.disintegrate, 100)
+    music.setVolume(10)
+    info.changeScoreBy(1)
+})
+scene.onHitTile(SpriteKind.Projectile, 1, function (sprite) {
+    sprite.destroy(effects.fire, 100)
+    info.changeLifeBy(-1)
+})
+controller.left.onEvent(ControllerButtonEvent.Pressed, function () {
+    Steve.setPosition(30, 100)
+})
+info.onLifeZero(function () {
+    game.over(true, effects.confetti)
+})
+controller.right.onEvent(ControllerButtonEvent.Pressed, function () {
+    Steve.setPosition(130, 100)
+})
+controller.up.onEvent(ControllerButtonEvent.Pressed, function () {
+    Steve.setPosition(60, 100)
+})
+controller.down.onEvent(ControllerButtonEvent.Pressed, function () {
+    Steve.setPosition(100, 100)
+})
+let right: Sprite = null
+let down: Sprite = null
+let up: Sprite = null
+let left: Sprite = null
+let Lane = 0
+let Steve: Sprite = null
```

Comment:
In all the snippets above ``Steve`` was lower cased, should match through the examples.

I'd have a small preference towards camelCasing all variable names in the examples -- most all of the other tutorials do, and it's nice to have things consistent / use a practice that will transition nicely to javascript eventually